docs/fiddles/ipc/pattern-1/main.js

```diff
@@ -1,28 +1,24 @@
 const {app, BrowserWindow, ipcMain} = require('electron')
 const path = require('path')
 
+function handleSetTitle (event, title) {
+  const webContents = event.sender
+  const win = BrowserWindow.fromWebContents(webContents)
+  win.setTitle(title)
+}
+
 function createWindow () {
   const mainWindow = new BrowserWindow({
     webPreferences: {
       preload: path.join(__dirname, 'preload.js')
     }
   })
-
-  ipcMain.on('set-title', (event, title) => {
-    const webContents = event.sender
-    const win = BrowserWindow.fromWebContents(webContents)
-    win.setTitle(title)
-  })
-
   mainWindow.loadFile('index.html')
 }
 
 app.whenReady().then(() => {
+  ipcMain.on('set-title', handleSetTitle)
   createWindow()
-
-  app.on('activate', function () {
```

Comment:
This shouldn't be removed.

Reply:
I normally would not remove this. However, this part of the code did not appear in the below explanations [here](https://www.electronjs.org/docs/latest/tutorial/ipc#1-listen-for-events-with-ipcmainon) like other explanations. (Sorry (very) for late reply)